src/main/java/core/basesyntax/supplier/FigureSupplier.java

```diff
@@ -0,0 +1,76 @@
+package core.basesyntax.supplier;
+
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.Figure;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_RANDOM_NUMBER = 100;
+    private static final int FIGURES_COUNT = 5;
+    private static final String DEFAULT_COLOR = Color.WHITE.name();
+    private static final int DEFAULT_RADIUS = 10;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int randomIndex = random.nextInt(FIGURES_COUNT);
+        switch (randomIndex) {
+            case 0:
+                return getRandomIsoscelesTrapezoid();
+            case 1:
+                return getRandomSquare();
+            case 2:
+                return getRandomRightTriangle();
+            case 3:
+                return getRandomRectangle();
+            case 4:
+                return getRandomCircle();
+            default:
+                return getDefaultFigure();
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(DEFAULT_COLOR, DEFAULT_RADIUS);
+    }
+
+    private Circle getRandomCircle() {
+        String randomColor = colorSupplier.getRandomColor();
+        double randomRadius = random.nextInt(MAX_RANDOM_NUMBER);
+        return new Circle(randomColor, randomRadius);
+    }
+
+    private Square getRandomSquare() {
+        String randomColor = colorSupplier.getRandomColor();
+        double randomSide = random.nextInt(MAX_RANDOM_NUMBER);
+        return new Square(randomColor, (int) randomSide);
+    }
```

Comment:
```suggestion
        String randomColor = colorSupplier.getRandomColor();
        int randomSide = random.nextInt(MAX_RANDOM_NUMBER);
        return new Square(randomColor,  randomSide);
    }
```

same for other places